packages/app/pages/[modelName]/[documentTitle]/index.tsx

```diff
@@ -79,6 +87,53 @@ const EditDocumentPage = ({
         refreshDataInPlace(router)
     }, [formData.state, refreshDataInPlace])
 
+    useEffect(() => {
+        const privileges = user.privilegesForModelAndWorkflowNode(
+            modelName,
+            model.workflow.currentNode
+        )
+        // @ts-expect-error
+        const { userActivation } = globalThis.navigator
+
+        let intervalId: null | ReturnType<typeof globalThis.setInterval>
+        let consecutiveErrorCount = 0
+
+        if (userActivation) {
+            handleCollaborators()
+            intervalId = globalThis.setInterval(handleCollaborators, 3000)
+        }
+
+        async function handleCollaborators() {
+            // No need to keep triggering a re-render / sending a network request if collaboration isn't working.
+            if (consecutiveErrorCount > 5) {
+                console.warn(
+                    `Collaboration featured are disabled for document ${documentTitle}.`
+                )
+                return globalThis.clearInterval(intervalId)
+            }
+
+            const collaborator = adaptUserToCollaborator(
+                user,
+                privileges,
+                userActivation.hasBeenActive,
+                userActivation.isActive
+            )
+
+            const [error, collaborators] = await updateCollaborators(
+                collaborator,
+                documentTitle,
+                modelName
+            )
+
+            consecutiveErrorCount = error ? consecutiveErrorCount + 1 : 0
+
+            setClientCollaborators(filterActiveUser(collaborators, user))
```

Comment:
`filter` returns an array, so if there are no active collaborators, we still keep the same data type.